Nos/Views/Components/Images/ImageViewer.swift

```diff
@@ -0,0 +1,198 @@
+                    .scaleEffect(scale)
+                    .offset(x: offset.width, y: offset.height)
+                    .gesture(
+                        DragGesture()
+                            .onChanged { value in
+                                withAnimation {
+                                    offset = CGSize(
+                                        width: lastOffset.width + value.translation.width * 1.5,
+                                        height: lastOffset.height + value.translation.height * 1.5
+                                    )
+                                }
+                            }
+                            .onEnded { value in
+                                withAnimation {
+                                    let isLargeSwipeDown: Bool
+                                    if scale == minScale {
+                                        isLargeSwipeDown = value.translation.height > 50
+                                    } else {
+                                        isLargeSwipeDown = value.translation.height > 200
+                                    }
+                                    if isLargeSwipeDown {
+                                        dismiss()
+                                    } else {
+                                        let predictedEndOffset = CGSize(
+                                            width: lastOffset.width + value.predictedEndTranslation.width,
+                                            height: lastOffset.height + value.predictedEndTranslation.height
+                                        )
+                                        offset = predictedEndOffset
+                                        keepImageOnScreen(geometry: geometry)
+                                        lastOffset = offset
+                                    }
+                                }
+                            }
+                            .simultaneously(
+                                with: TapGesture(count: 2)
+                                    .onEnded {
+                                        withAnimation {
+                                            if scale == minScale {
+                                                scale = 3.0
+                                            } else {
+                                                scale = minScale
+                                                offset = .zero
+                                                lastOffset = .zero
+                                            }
+                                        }
+                                    }
+                            )
+                    )
+                    .frame(width: geometry.size.width, height: geometry.size.height)
+                    .clipped()
+                    .ignoresSafeArea()
+            }
+
+            ZStack(alignment: .topLeading) {
+                Color.clear
+
+                Button(
```

Comment:
Given that Button is a well-known component in SwiftUI, I prefer the more succinct declaration:

```
Button {
    dismiss()
} label: {
    ....
}
```